core/gen_index.ts

```diff
@@ -62,7 +62,7 @@ projectConfig.schemaSuffix = "${
   }" || projectConfig.schemaSuffix;
 
 // Merge in general project config overrides.
-projectConfig = { ...projectConfig, ...${projectOverridesJsonString} };
+projectConfig = { ...projectConfig, ...${projectOverridesJsonString}, vars: { ...projectConfig.vars, ...${projectOverridesJsonString}.vars } };
```

Comment:
Why not deepmerge here as well?

Reply:
This is effectively the index.js that we generate for each DF project. Plumbing through dependencies on other NPM packages at this stage is a bit of a pain, while this isn't ideal, I think it's the best option we have for now.